Assignment 2.Rmd

```diff
@@ -81,20 +101,137 @@ plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender)
+D1 <- read.csv("~/R/HUDK/assignment2/hudk4050-classes.csv")
+
+#rename rows with student names and drop student names
+rownames(D1) <- paste(D1$First.Name,D1$Last.Name)
+D1 <- D1[,3:8] 
+
+#standardize space/no space in how subjects are refered to and make it a string
+D1[,1] <- gsub(" ", "", as.character(D1[,1]), fixed = TRUE)
+D1[,2] <- gsub(" ", "", as.character(D1[,2]), fixed = TRUE)
+D1[,3] <- gsub(" ", "", as.character(D1[,3]), fixed = TRUE)
+D1[,4] <- gsub(" ", "", as.character(D1[,4]), fixed = TRUE)
+D1[,5] <- gsub(" ", "", as.character(D1[,5]), fixed = TRUE)
+D1[,6] <- gsub(" ", "", as.character(D1[,6]), fixed = TRUE)
+
+#correct the data of one of our friends(there were only numbers, we need to input letters) 
+D1["ZIMO CHEN",]
+D1["ZIMO CHEN",1] <- gsub(" ", "", paste("HUDK",D1["ZIMO CHEN",1]), fixed = TRUE)
+D1["ZIMO CHEN",2] <- gsub(" ", "", paste("HUDM",D1["ZIMO CHEN",2]), fixed = TRUE)
+D1["ZIMO CHEN",3] <- gsub(" ", "", paste("HUDM",D1["ZIMO CHEN",3]), fixed = TRUE)
+D1["ZIMO CHEN",4] <- gsub(" ", "", paste("HUDM",D1["ZIMO CHEN",4]), fixed = TRUE)
+D1["ZIMO CHEN",]
+
+#also, lets remove HUDK4050 - we already know we are taking that class together (and it would mess up our graphs and our centrality measures!)
+D1[,1][D1[,1] == "HUDK4050"] <- ""
+D1[,2][D1[,2] == "HUDK4050"] <- ""
+D1[,3][D1[,3] == "HUDK4050"] <- ""
+D1[,4][D1[,4] == "HUDK4050"] <- ""
+D1[,5][D1[,5] == "HUDK4050"] <- ""
+D1[,6][D1[,6] == "HUDK4050"] <- ""
+
+```
+
+Now let's create the person-class matrix
+
+```{r}
+library(tidyr)
+# here is our list of unique course names
+courses <- sort((unique(c(D1[,1],D1[,2],D1[,3],D1[,4],D1[,5],D1[,6]))))
+
+#let's get rid of the (missing) "" that's on our list
+courses <- courses[2:length(courses)]
+
+# here goes our matrix structure
+PC <- matrix(0,length(D1[,1]),length(courses))
+rownames(PC) <- rownames(D1)
+colnames(PC) <- courses
+
+# let's fill in the values for who went to each course (and each of the six class columns of D1)
+for(i in 1:length(courses)){
+  for(j in 1:6){
+    PC[,i] <- PC[,i] + as.integer(D1[,j] == courses[i])
+  }
+}
+
+```
+
+Now we build our person-person matrix and plot the network:
+```{r}
+#we do the trick of multiplying PC by its transpose...
+PP <- PC%*%t(PC)
+
+# now create the graph object
+g <- graph_from_incidence_matrix(PP, directed = FALSE, mode = c("all", "out", "in", "total"), multiple = FALSE, weighted = TRUE, add.names = NULL)
+
+# and plot it
+plot(g,layout=layout.fruchterman.reingold, vertex.label=NA, vertex.size=7)
+
+```
+
+
+Finally, betweeness centrality and degree
+```{r}
+#betweeness
+btness <- betweenness(g, v = V(g), directed = FALSE, weights = NULL, nobigint = TRUE, normalized = FALSE)
+
+#person wih the highest "betweeness centrality"
+btness[btness==max(btness)]
+
+#centrality degree
+degree <- degree(g)
+#person wih highest degree measure
+degree[degree==max(degree)]
+```
+
+So Bernell Downer (betweness) and Yujun Zhang (degree) are our most central persons!
+
```

Comment:
Good job! It seems you work very hard and figure out ways to complete this! Keep up the good work! Maybe next time you want to think more about how to make the code more concise.